src/WebJobs.Script.Scaling/AzureWorkerTable.cs

```diff
@@ -0,0 +1,230 @@
+            query.Where(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, AzureEnvironments.WorkerPartitionKey));
+
+            var table = await EnsureWorkerTable();
+            var results = await ExecuteQueryAsync(table, query);
+            return results.OfType<IWorkerInfo>();
+        }
+
+        public async Task<IWorkerInfo> GetManager()
+        {
+            var table = await EnsureWorkerTable();
+
+            // read manager row
+            var operation = TableOperation.Retrieve<AzureWorkerInfo>(AzureEnvironments.ManagerPartitionKey, AzureEnvironments.ManagerRowKey);
+            var result = await table.ExecuteAsync(operation);
+            var manager = result.Result as IWorkerInfo;
+            if (manager != null)
+            {
+                // read from worker table
+                var partitionKey = AzureEnvironments.WorkerPartitionKey;
+                var rowKey = AzureEnvironments.GetWorkerRowKey(manager.StampName, manager.WorkerName);
+                operation = TableOperation.Retrieve<AzureWorkerInfo>(partitionKey, rowKey);
+                result = await table.ExecuteAsync(operation);
+                manager = result.Result as IWorkerInfo;
+            }
+
+            return manager;
+        }
+
+        public async Task SetManager(IWorkerInfo worker)
+        {
+            // update manager row
+            var entity = new AzureWorkerInfo
+            {
+                PartitionKey = AzureEnvironments.ManagerPartitionKey,
+                RowKey = AzureEnvironments.ManagerRowKey,
+                StampName = worker.StampName,
+                WorkerName = worker.WorkerName,
+                ETag = "*"
+            };
+
+            var operation = TableOperation.InsertOrReplace(entity);
+            var table = await EnsureWorkerTable();
+            await table.ExecuteAsync(operation);
+        }
+
+        private async Task<CloudTable> EnsureWorkerTable()
+        {
+            if (_workerTable == null)
+            {
+                var account = CloudStorageAccount.Parse(AzureEnvironments.StorageConnectionString);
+                var client = account.CreateCloudTableClient();
+                var table = client.GetTableReference(AzureWorkerTableName);
+                await table.CreateIfNotExistsAsync();
+                _workerTable = table;
+            }
+
+            return _workerTable;
+        }
+
+        private async Task<CloudBlockBlob> EnsureBlobLease()
+        {
+            if (_blobLease == null)
+            {
+                var account = CloudStorageAccount.Parse(AzureEnvironments.StorageConnectionString);
+                var client = account.CreateCloudBlobClient();
+                var container = client.GetContainerReference(AzureWorkerTableName);
+                await container.CreateIfNotExistsAsync();
+                var blobLease = container.GetBlockBlobReference(AzureEnvironments.SiteName);
```

Comment:
This is the runtime site name which is unique for site and each slots.   BTW, the slotname can be a little off during swap transition.   Since we have to warm up the PROD slot before redirect traffic to it, there is a period of time where prod slot may still serving staging slot http request.